src/pinns_pde_solve.jl

```diff
@@ -1,5 +1,68 @@
-struct PhysicsInformedNN{D}
+"""
+Algorithm for solving Physics Informed Neural Networks problem.
+
+Arguments:
+* `dx` is discretization of grid
+* `chain` is a Flux.jl chain with d dimensional input and 1 dimensional output,
+* `init_params` is the initial parameter of the neural network,
+* `phi` is trial solution,
+* `autodiff` is a boolean variable that determines whether to use automatic, differentiation(not supported while) or numerical,
+* `derivative` is method that calculate derivative,
+* `training_strategies` is determines which training strategy will be used.
+"""
+
+struct PhysicsInformedNN{D,C,P,PH,DER,T,K}
   dx::D
+  chain::C
+  initθ::P
+  phi::PH
+  autodiff::Bool
+  derivative::DER
+  training_strategies::T
+  kwargs::K
+end
+
+function PhysicsInformedNN(dx,
+                           chain,
+                           init_params = nothing;
+                           _phi = nothing,
+                           autodiff=false,
+                           _derivative = nothing,
+                           training_strategies = TrainingStrategies(),
+                           kwargs...)
+    if init_params === nothing
+        if chain isa FastChain
+            initθ = DiffEqFlux.initial_params(chain)
+        else
+            initθ,re  = Flux.destructure(chain)
+        end
+    else
+        initθ = init_params
+    end
+    isuinplace = chain.layers[end].out == 1
+    dim = chain.layers[1].in
+
+    if _phi == nothing
+        phi = get_phi(chain,isuinplace)
+    else
+        phi = _phi
+    end
+
+    if _derivative == nothing
+        derivative = get_derivative(dim,phi,autodiff,isuinplace)
+    else
+        derivative = _derivative
+    end
+
+    PhysicsInformedNN(dx,chain,initθ,phi,autodiff,derivative, training_strategies, kwargs)
+end
+
+struct TrainingStrategies
+    stochastic_loss::Bool
+end
+
+function TrainingStrategies(;stochastic_loss=true)
```

Comment:
instead of true false, make it `strategy = GridTraining()`, or `strategy = StochasticTraining()`, I think we have a list of 5 we want to do now, and that will only grow, so booleans won't manage it well and instead of a dispatch-based way to writing each loss function will be needed.